composer/algorithms/swa/swa.py

```diff
@@ -96,22 +133,22 @@ def apply(self, event: Event, state: State, logger: Logger) -> None:
                 state.optimizers[0],
                 swa_lr=self.swa_lr,
                 anneal_epochs=self.anneal_epochs,
-                anneal_strategy='cos',
+                anneal_strategy=self.anneal_strategy,
             )
 
         if self.swa_model is None:
             self.swa_model = AveragedModel(state.model)
 
         self.swa_model.update_parameters(state.model)  # type: ignore
 
-        if self.swa_scheduler is None:
-            raise ValueError('SWA LR scheduler was not set.')
-        self.swa_scheduler.step()
-
-        ## end of training
-        if float(state.get_elapsed_duration()) >= 1.0:
-            device = next(self.swa_model.parameters()).device
-            # TODO(laura) this does not apply the batch split fn. This may result in cuda OOM
-            update_bn(state.train_dataloader, model=self.swa_model, device=device)
-            state.model = self.swa_model
-            log.info('Updated BN and set model to the averaged model')
+        if self.schedule_swa_lr:
+            if self.swa_scheduler is None:
+                raise ValueError('SWA LR scheduler was not set.')
+            self.swa_scheduler.step()
+
+        # End of swa window
+        if float(state.get_elapsed_duration()) >= self.swa_end:
+            assert type(self.swa_model.module) == type(state.model)
```

Comment:
Should this be a ValueError or RuntimeError? (I presume this check is for deepspeed?)

Reply:
I'm actually not sure why this is here. It was a pre-existing part of the code, and I believe it predates deepspeed

Reply:
yeah that's an odd check.. sound like ti should be part of a `test_swa` instead of in this code?

Reply:
in my original implementation this check was needed to make sure we did not change the type of the Model to AveragedModel (from torch.optim.swa_utils.AveragedModel)